bmc-log-collector/log-collector.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"os"
+	"path"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type SystemEventLog struct {
+	Od_Id             string   `json:"@odata.id"`
+	Od_Type           string   `json:"@odata.type"`
+	Create            string   `json:"Created"`
+	Description       string   `json:"Description"`
+	EntryCode         string   `json:"EntryCode"`
+	EntryType         string   `json:"EntryType"`
+	GeneratorId       string   `json:"GeneratorId"`
+	Id                string   `json:"Id"`
+	Message           string   `json:"Message"`
+	MessageArgs       []string `json:"MessageArgs"`
+	OdCnt_MessageArgs int      `json:"MessageArgs@odata.count"`
+	MessageId         string   `json:"MessageId"`
+	Name              string   `json:"Name"`
+	SensorNumber      int      `json:"SensorNumber"`
+	SensorType        string   `json:"SensorType"`
+	Severity          string   `json:"Severity"`
+	Serial            string
+	NodeIP            string
+	BmcIP             string
+}
+
+type Redfish struct {
+	Name        string           `json:"Name"`
+	Count       int              `json:"Members@odata.count"`
+	Context     string           `json:"@odata.context"`
+	Id          string           `json:"@odata.id"`
+	Type        string           `json:"@odata.type"`
+	Description string           `json:"Descriptionta"`
+	Sel         []SystemEventLog `json:"Members"`
+}
+
+type logCollector struct {
+	machinesPath string       // Machine list path
+	rfUrl        string       // Redfish API address
+	ptrDir       string       // Pointer store
+	testMode     bool         // when testMode is true, write text file for test
```

Comment:
Test mode flag is not recommended.
It is better to prepare an interfcae for logCollector and logWriter,  then plug the stub from the outside of function.

Reply:
Eliminate testMode and use the log writer module to detect test mode.